pkg/skaffold/trigger/triggers_test.go

```diff
@@ -222,33 +179,27 @@ func TestStartTrigger(t *testing.T) {
 	tests := []struct {
 		description string
 		trigger     Trigger
+		mockWatch   func(string, chan<- notify.EventInfo, ...notify.Event) error
 	}{
 		{
 			description: "fsNotify trigger works",
-			trigger: &fsNotifyTrigger{
-				Interval:   200 * time.Millisecond,
-				workspaces: nil,
-				isActive:   func() bool { return false },
-				watchFunc: func(string, chan<- notify.EventInfo, ...notify.Event) error {
-					return nil
-				},
+			trigger:     fsNotify.New(nil, func() bool { return false }, 1),
+			mockWatch: func(string, chan<- notify.EventInfo, ...notify.Event) error {
+				return nil
 			},
 		},
 		{
 			description: "fallback on polling trigger",
-			trigger: &fsNotifyTrigger{
-				Interval:   200 * time.Millisecond,
-				workspaces: nil,
-				isActive:   func() bool { return false },
-				watchFunc: func(string, chan<- notify.EventInfo, ...notify.Event) error {
-					return fmt.Errorf("failed to start watch trigger")
-				},
+			trigger:     fsNotify.New(nil, func() bool { return false }, 1),
+			mockWatch: func(string, chan<- notify.EventInfo, ...notify.Event) error {
+				return fmt.Errorf("failed to start watch trigger")
 			},
 		},
 	}
 
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
+			t.Override(&fsNotify.Watch, test.mockWatch)
```

Comment:
Is this working as intended? Wouldn't the trigger struct already have it's `watchFunc` set before this override happens? Since the `watchFunc` is grabbed from `fsNotify.Watch` upon calling `fsNotify.New`

Reply:
Thanks! fixed it.